app/main.py

```diff
@@ -1,2 +1,51 @@
+from copy import deepcopy
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.table = [[] for _ in range(8)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        threshold = int(self.capacity * self.load_factor)
+        if self.length == threshold:
+            self.resize()
+
+        hash_ = hash(key)
+        index = hash_ % self.capacity
+
+        while True:
+            if not self.table[index]:
+                self.table[index] = [key, value, hash_]
+                self.length += 1
+                return
+
+            if self.table[index][0] == key and self.table[index][2] == hash_:
```

Comment:
compare hashes first
comparing integers is faster than comparing complex keys, e.x. long tuples

Reply:
fix everywhere